File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums.sort()
        count = 0
        currNum = nums[0]

        for num in nums:
            if num != currNum:
                count = 1
                currNum = num
            else:
                count += 1

            if count > (len(nums)/2):
                return currNum

File: 169-majority-element/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    nums.sort((a, b) => a - b)
    const n = nums.length / 2;
    let count = 0;
    let currNum = nums[0]

    for (const num of nums){
        if (num !== currNum) {
            count = 1
            currNum = num
        } else {
            count++;
        }

        if (count > n) return currNum
    }
};